packages/core/src/tools/valueHistory.spec.ts

```diff
@@ -54,6 +54,15 @@ describe('valueHistory', () => {
 
       expect(valueHistory.find(15 as RelativeTime)).toBeUndefined()
     })
+
+    describe('with predicate', () => {
+      it('should return the value of the entry matching the startTime and the predicate', () => {
+        valueHistory.add('foo', 0 as RelativeTime).close(10 as RelativeTime)
+        valueHistory.add('bar', 20 as RelativeTime)
+
+        expect(valueHistory.find(15 as RelativeTime, (entry) => entry.endTime === 10)).toEqual('foo')
+      })
```

Comment:
<!-- Warning represent user-facing problems. If possible, it's great to follow this kind of comment with a suggestion.  -->
**❗ warning:** ‏given your strategy with `() => true` predicate for logs session, I would expect:

```
expect(valueHistory.find(15 as RelativeTime, () => true)).toEqual('bar')
```

but currently it returns `'foo'`.

am I missing something?

Reply:
To me it's correct. It returns 'foo' because the "foo" startTime is 15 which is greater than 0 but lower than 20. 
In the context of logs it means that we return the latest session even if closed.

Reply:
work as expected with 25, so it looks good 😅